### PR TITLE
[CARBONDATA-440] fixing no kettle issue for IUD.

### PR DESCRIPTION
**Pull request by ravikiran23**
For iud data load flow will be used. so in the case of NO-KETTLE, need to handle data load.

load count/ segment count should be string because in compaction case it will be 2.1

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.5.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/559/

File: integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala [hunk 1]
```diff
@@ -719,16 +720,51 @@ object CarbonDataRDDFactory {
...
+                    new RddIteratorForUpdate(iter, carbonLoadModel))
+                  carbonLoadModel.setRddIteratorKey(rddIteratorKey)
+                  CarbonDataLoadForUpdate
+                    .run(carbonLoadModel, index, storePath, kettleHomePath,
+                      segId, loadMetadataDetails, executionErrors)
+                } finally {
+                  RddInpututilsForUpdate.remove(rddIteratorKey)
+                }
+              }
+              else {
```
Comment (jackylk) [inline integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala]: move to previous line

File: integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala [hunk 2]
```diff
@@ -719,16 +720,51 @@ object CarbonDataRDDFactory {
               loadMetadataDetails.setLoadStatus(CarbonCommonConstants.STORE_LOADSTATUS_SUCCESS)
               val rddIteratorKey = CarbonCommonConstants.RDDUTIL_UPDATE_KEY +
                                    UUID.randomUUID().toString
+              if (useKettle) {
```
Comment (jackylk) [inline integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala]: how about in carbon-spark2 module, can you check the same in that module also?

**Comment by jackylk** [comment]
I verified with `mvn clean verify -Pno-kettle -Pspark-1.6` but it failed in test case `insert from hive-sum expression`

File: integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala (hunk 2 above)
Comment (ravikiran23) [inline integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala]: as of now IUD is supported in 1.6.2. support is not there for 2.1

File: integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala (hunk 1 above)
Comment (ravikiran23) [inline integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala]: fixed

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/565/

**Comment by ravikiran23** [comment]
@jackylk  i verified the same test case with new code with out my fix , it is still failing.  this may be due to some other PR. my code doesnt impact insert into flow.

**Comment by ravipesala** [comment]
@jackylk Please review and merge this PR, I will fix the testcases for no kettle flow and raise in another PR

**Comment by jackylk** [comment]
LGTM

File: integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala [hunk 3]
```diff
@@ -719,16 +720,50 @@ object CarbonDataRDDFactory {
...
+                try {
```
Comment (jackylk) [inline integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala]: move `try` to `CarbonDataLoadForUpdate.run` only, we should limit the try scope, do the same for next `try` also.

File: integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala [hunk 4]
```diff
@@ -719,16 +720,50 @@ object CarbonDataRDDFactory {
...
+                    null,
+                    null,
+                    segId,
+                    loadMetadataDetails)
+                  // Intialize to set carbon properties
+                  loader.initialize()
+
+                  loadMetadataDetails.setLoadStatus(CarbonCommonConstants.STORE_LOADSTATUS_SUCCESS)
+                  new DataLoadExecutor()
+                    .execute(carbonLoadModel, loader.storeLocation, recordReaders.toArray)
```
Comment (jackylk) [inline integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala]: move to previous line, break the line at parameter list

File: integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala [hunk 5]
```diff
@@ -719,16 +720,50 @@ object CarbonDataRDDFactory {
...
+
+                } catch {
+                  case e: BadRecordFoundException =>
+                    loadMetadataDetails
+                      .setLoadStatus(CarbonCommonConstants.STORE_LOADSTATUS_PARTIAL_SUCCESS)
```
Comment (jackylk) [inline integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala]: move to previous line, break the line at parameter list

File: integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala [hunk 6]
```diff
@@ -719,16 +720,50 @@ object CarbonDataRDDFactory {
               loadMetadataDetails.setLoadStatus(CarbonCommonConstants.STORE_LOADSTATUS_SUCCESS)
               val rddIteratorKey = CarbonCommonConstants.RDDUTIL_UPDATE_KEY +
                                    UUID.randomUUID().toString
+              if (useKettle) {
+                try {
+                  RddInpututilsForUpdate.put(rddIteratorKey,
+                    new RddIteratorForUpdate(iter, carbonLoadModel))
+                  carbonLoadModel.setRddIteratorKey(rddIteratorKey)
+                  CarbonDataLoadForUpdate
+                    .run(carbonLoadModel, index, storePath, kettleHomePath,
```
Comment (jackylk) [inline integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala]: move to previous line, break the line at parameter list

File: integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala [hunk 7]
```diff
@@ -719,16 +720,50 @@ object CarbonDataRDDFactory {
...
+                  val serializer = SparkEnv.get.closureSerializer.newInstance()
+                  var serializeBuffer: ByteBuffer = null
+                    recordReaders += new CarbonIteratorImpl(
+                      new NewRddIterator(iter,
+                        carbonLoadModel,
+                        TaskContext.get()))
+
+                  val loader = new SparkPartitionLoader(carbonLoadModel,
+                    index,
+                    null,
```
Comment (jackylk) [inline integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala]: You are following different code style, can you make the style like other code?

**Comment by ravipesala** [comment]
@ravikiran23 Please work on comments given by @jackylk , we should merge this soon otherwise it will be   blocking for testing.

File: integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala (hunk 3 above)
Comment (ravikiran23) [inline integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala]: fixed

File: integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala (hunk 4 above)
Comment (ravikiran23) [inline integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala]: fixed

File: integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala (hunk 5 above)
Comment (ravikiran23) [inline integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala]: fixed

File: integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala (hunk 7 above)
Comment (ravikiran23) [inline integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala]: fixed

File: integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala (hunk 6 above)
Comment (ravikiran23) [inline integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala]: fixed

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/618/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/619/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/620/

**Comment by ravikiran23** [comment]
@jackylk  please merge this PR. the test case which is failing is insert into test case. this is not because of my PR.
i ve handled the comments.

**Comment by jackylk** [comment]
LGTM

**Pull request closed without merging**